### DISCUSSION TITLE
Support for Auto-Generating PHP Types from `.graphql` Files and Handling Split Schemas

### DISCUSSION THREAD

**Discussion by e-labInnovations** [webonyx/graphql-php#1596] [Q&A]
Hello team,

I'm working on a project where I use `webonyx/graphql-php` to implement a GraphQL server. My schema is organized using `.graphql` files that define various types, queries, and mutations. I aim to streamline our development process by auto-generating the corresponding PHP classes (like `QueryType.php`, `MutationType.php`, etc.) from these `.graphql` files.

### My questions are:

1. **Is there any built-in support in `webonyx/graphql-php` for auto-generating PHP types from `.graphql` schema files?**
   - If not, is this something that might be considered for future development?

2. **Are there any recommended third-party tools or best practices within the PHP ecosystem that can help achieve this?**
   - Specifically, we're looking for a way to generate classes that include only the skeleton code, allowing developers to focus on implementing resolver logic.

3. **Challenges with Using `schema.graphql` Directly:**
   - When using the `schema.graphql` file directly in the project, as shown in the [official examples](https://github.com/webonyx/graphql-php/tree/master/examples/02-schema-definition-language), we face limitations in splitting the schema across multiple files. This makes it difficult to manage larger projects where modularity is crucial. 
   - **Is there a recommended approach to handle split schemas in `webonyx/graphql-php`?** 

4. **Custom CLI Tool Consideration:**
   - If auto-generation is not currently supported, would building a custom CLI tool or script be the best approach? 
   - Are there any guidelines or existing libraries that could assist in parsing `.graphql` files and generating the necessary PHP classes while addressing the modularity issue?

Any insights or recommendations on approaching these challenges within the `webonyx/graphql-php` framework would be greatly appreciated!

Thank you for your time and for providing such a powerful GraphQL library.

**Comment by Warxcell** [comment]
1. Yes, I did a package myself for my needs - https://github.com/Warxcell/graphql-codegen
2. :arrow_up: 
3. Yes - Again my package do this https://github.com/Warxcell/graphql-bundle - I'm using exactly same approach - split project by modules - each module brings its own schema.

There is no documentation, but I'm happy to assist.
